presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -170,13 +172,15 @@ public static long parseTimestampWithTimeZone(TimeZoneKey timeZoneKey, String ti
     }
 
     /**
-     * Parse date-time without zone.
+     * Parse date-time with or without zone. If {@code value} specifies zone, it will be discarded,
+     * as it has no impact on the returned value.
      *
-     * @return timestamp
+     * @return timestamp component of the value
      */
     public static long parseTimestampWithoutTimeZone(String value)
     {
-        return TIMESTAMP_WITHOUT_TIME_ZONE_FORMATTER.parseMillis(value);
+        LocalDateTime localDateTime = TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.parseLocalDateTime(value);
+        return localDateTime.toDateTime(UTC).getMillis();
```

Comment:
All we need is `org.joda.time.LocalDateTime#getLocalMillis`, but it's not accessible. @haozhun any better idea than what i have here?